akka-javasdk/src/main/scala/akka/javasdk/impl/agent/AgentImpl.scala

```diff
@@ -124,48 +124,49 @@ private[impl] final class AgentImpl[A <: Agent](
     new ReflectiveAgentRouter[A](factory, componentDescriptor.methodInvokers, serializer)
   }
 
-  override def handleCommand(command: SpiAgent.Command): Future[SpiAgent.Effect] = {
-
-    // FIXME(tracing): add full telemetry context to SpiAgent.Command
-    val telemetryContext = command.span.map(OtelContext.root.`with`)
-    command.span.foreach(s => MDC.put(Telemetry.TRACE_ID, s.getSpanContext.getTraceId))
-    // smuggling 0 arity methods called from the component client through here
-    val cmdPayload = command.payload.getOrElse(BytesPayload.empty)
-    val metadata: Metadata = MetadataImpl.of(command.metadata)
-    val agentContext = new AgentContextImpl(sessionId, regionInfo.selfRegion, metadata, telemetryContext, tracerFactory)
-
-    try {
-      // we need the agent at this scope.
-      // therefore, we initialize it exceptionally outside the router
-      val agent = factory(agentContext)
-      val commandEffect = router.handleCommand(agent, command.name, cmdPayload, agentContext)
-
-      def primaryEffect =
-        commandEffect match {
-          case e: AgentEffectImpl       => e.primaryEffect
-          case e: AgentStreamEffectImpl => e.primaryEffect
-        }
+  override def handleCommand(command: SpiAgent.Command): Future[SpiAgent.Effect] =
+    Future {
```

Comment:
why do we need this? this should have been taken care of by the runtime when calling the spi, by the `wrapSpiComponents`

Reply:
You are right, but somehow we are still observing a wrong behavior. Wrapping in a future is fixing it for now, but we need to better understand what is going on.